### PR TITLE
CORTX-32718: CSM Setup logs improved

### PR DESCRIPTION
**Pull request by abhijit1patil**
Signed-off-by: abhijit1patil <abhijit.1.patil@seagate.com>

# Pull Request
## Problem Statement
-   [CORTX-32718 improve setup logs](https://jts.seagate.com/browse/CORTX-32718)

## Design
-  Add logs in all phases of the CSM Setup

-   \[ \] Coding conventions are followed and code is consistent

## Custom Build and Deployment
- [Custom Build Job](https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/7260/)
- Custom Build
   cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-7260-custom-ci
   cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-7260-custom-ci
   cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-7260-custom-ci
- [Deployment Job](url)

-   \[ \] Unit and System Tests are added
-   \[ \] Test Cases cover Happy Path, Non-Happy Path and Scalability
-   \[ \] Testing was performed with RPM

-   \[ \] Interface change (if any) are documented
-   \[ \] Side effects on other features (deployment/upgrade)
-   \[ \] Dependencies on other component(s)

-   \[ \] JIRA number/GitHub Issue added to PR
-   \[ \] PR is self reviewed
-   \[ \] Jira and state/status is updated and JIRA is updated with PR link
-   \[ \] Check if the description is clear and explained

-   \[ \] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: csm/conf/configure.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ async def execute(self, command):
         :param command:
         :return:
         """
+        Log.info("csm_setup: configure phase started.")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:53]: csm_setup -> Setup. These are CSM setup logs. No need to mention csm_setup.

configure phase started. -> Initiating Config phase.

File: csm/conf/configure.py [hunk 2]
```diff
@@ -110,9 +111,11 @@ async def execute(self, command):
                        f"{e} - {str(traceback.format_exc())}")
             Log.error(err_msg)
             raise CsmSetupError(err_msg)
+        Log.info("csm_setup: configure phase completed.")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:114]: csm_setup -> Setup. Change this at all the places.

configure phase completed. -> Successfully passed Config phase.

File: csm/conf/configure.py [hunk 3]
```diff
@@ -110,9 +111,11 @@ async def execute(self, command):
...
         return Response(output=const.CSM_SETUP_PASS, rc=CSM_OPERATION_SUCESSFUL)
 
     def _prepare_and_validate_confstore_keys(self):
+        Log.info("Preparing and validating configuration store keys")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:118]: Preparing and validating configuration store keys -> Config: Validating required configuration.

File: csm/conf/configure.py [hunk 4]
```diff
@@ -137,7 +140,7 @@ def create(self):
 
     @staticmethod
     def _set_csm_endpoint():
-        Log.info("Setting csm endpoint in csm config")
+        Log.info("Setting csm endpoint in csm configuration")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:143]: csm -> CSM
Setting csm endpoint in csm configuration -> Setting CSM endpoint in configuration.
Prepend all the info logs with the phase from where they are origination. 
Like Config: Setting CSM endpoint in configuration.

File: csm/conf/configure.py [hunk 5]
```diff
@@ -153,6 +156,7 @@ def _set_csm_endpoint():
 
     @staticmethod
     def _set_s3_endpoints():
+        Log.info("Setting S3 endpoints in csm config")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:159]: Setting S3 endpoints in csm config -> Setting S3 endpoints in configuration. 
Be consistent with the words you are choosing. Like config -> configuration

File: csm/conf/configure.py [hunk 6]
```diff
@@ -177,7 +181,7 @@ def _set_s3_info():
         This Function will set s3  related configurations.
         :return:
         """
-        Log.info("Setting S3 configurations in csm config")
+        Log.info("Setting S3 information in csm config")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:184]: Same here as above

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]
Please make sure that  grammar is proper. Make use of capital letters and in-corporate the changes requested in all the files.

File: csm/conf/configure.py (hunk 1 above)
Comment (abhijit1patil) [reply csm/conf/configure.py]: Updated

File: csm/conf/configure.py (hunk 2 above)
Comment (abhijit1patil) [reply csm/conf/configure.py]: updated

File: csm/conf/configure.py (hunk 3 above)
Comment (abhijit1patil) [reply csm/conf/configure.py]: updated

File: csm/conf/configure.py (hunk 4 above)
Comment (abhijit1patil) [reply csm/conf/configure.py]: updated

File: csm/conf/configure.py [hunk 7]
```diff
@@ -50,17 +50,17 @@ async def execute(self, command):
...
         """
+        Log.info("Setup: Initiating Config phase.")
         try:
             Conf.load(const.CONSUMER_INDEX, command.options.get(const.CONFIG_URL))
             Setup.setup_logs_init()
-            Log.info("Executing csm_setup: configure phase.")
             Setup.load_csm_config_indices()
         except KvError as e:
-            Log.error(f"Configuration Loading Failed {e}")
+            Log.error(f"Setup: Configuration loading failed {e}")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:59]: Setup -> Config

File: csm/conf/configure.py [hunk 8]
```diff
@@ -227,19 +231,19 @@ async def _set_unsupported_feature_info(self):
         """
 
         def get_component_list_from_features_endpoints():
-            Log.info("Get Component List.")
+            Log.info("Config: Getting component list.")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:234]: Comment all the functions and functionalities related to unsupported features

File: csm/conf/configure.py [hunk 9]
```diff
@@ -227,19 +231,19 @@ async def _set_unsupported_feature_info(self):
...
             feature_endpoints = Json(
                 const.FEATURE_ENDPOINT_MAPPING_SCHEMA).load()
             component_list = [feature for v in feature_endpoints.values() for
                               feature in v.get(const.DEPENDENT_ON)]
             return list(set(component_list))
         try:
-            Log.info("Set unsupported feature list to ES.")
+            Log.info("Config: Setting unsupported feature list to ES.")
```
Comment (pawankumarsrivastava) [inline csm/conf/configure.py:241]: ES is not in picture now. Anyways comment this code

File: csm/conf/init.py [hunk 10]
```diff
@@ -35,12 +35,12 @@ async def execute(self, command):
...
         :return:
         """
+        Log.info("Setup: Initiating Init phase.")
         try:
             Conf.load(const.CONSUMER_INDEX, command.options.get(const.CONFIG_URL))
             Setup.setup_logs_init()
-            Log.info("Executing csm_setup: init phase.")
         except KvError as e:
-            Log.error(f"Configuration Loading Failed {e}")
+            Log.error(f"Setup: Configuration Loading Failed {e}")
```
Comment (pawankumarsrivastava) [inline csm/conf/init.py:43]: Setup -> Init

File: csm/conf/prepare.py [hunk 11]
```diff
@@ -43,13 +43,13 @@ async def execute(self, command):
...
         """
+        Log.info("Setup: Initiating Prepare phase.")
         try:
             Conf.load(const.CONSUMER_INDEX, command.options.get(const.CONFIG_URL))
             Setup.setup_logs_init()
-            Log.info("Executing csm_setup: prepare phase.")
             Setup.load_csm_config_indices()
         except KvError as e:
-            Log.error(f"Configuration Loading Failed {e}")
+            Log.error(f"Setup: Configuration Loading Failed {e}")
```
Comment (pawankumarsrivastava) [inline csm/conf/prepare.py:52]: Setup -> Prepare

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/conf/configure.py (hunk 9 above)
Comment (rohitkolapkar) [reply csm/conf/configure.py:241]: please change ES to storage/database and yes comment this code as asked by Pawan

File: csm/conf/configure.py (hunk 7 above)
Comment (rohitkolapkar) [inline csm/conf/configure.py:59]: Setup -> config ?

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]

File: csm/conf/prepare.py (hunk 11 above)
Comment (abhijit1patil) [reply csm/conf/prepare.py:52]: Updated

File: csm/conf/init.py (hunk 10 above)
Comment (abhijit1patil) [reply csm/conf/init.py:43]: updated

File: csm/conf/configure.py (hunk 7 above)
Comment (abhijit1patil) [reply csm/conf/configure.py:59]: updated

File: csm/conf/configure.py (hunk 9 above)
Comment (abhijit1patil) [reply csm/conf/configure.py]: updated

File: csm/conf/configure.py (hunk 7 above)
Comment (abhijit1patil) [reply csm/conf/configure.py]: updated

File: csm/conf/configure.py (hunk 8 above)
Comment (pawankumarsrivastava) [reply csm/conf/configure.py]: @abhijit1patil Is this done?

Comment (abhijit1patil) [reply csm/conf/configure.py]: Updated

File: csm/core/controllers/unsupported_features.py [hunk 12]
```diff
@@ -18,7 +18,7 @@
 from csm.core.blogic import const
 
 
-@CsmView._app_routes.view("/api/v2/unsupported_features")
+# @CsmView._app_routes.view("/api/v2/unsupported_features")
```
Comment (rohitkolapkar) [inline csm/core/controllers/unsupported_features.py:21]: If you are commenting this from routes.py then no need to comment it here. Please check once.

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]

File: csm/core/controllers/unsupported_features.py (hunk 12 above)
Comment (abhijit1patil) [reply csm/core/controllers/unsupported_features.py:21]: Make sense 
I will uncomment this

**Comment by abhijit1patil** [comment]
2022-07-18 07:43:03 csm_setup [45]: INFO [execute] Setup:  Initiating Post Install phase.
2022-07-18 07:43:03 csm_setup [45]: INFO [load_csm_config_indices] Loading CSM configuration
2022-07-18 07:43:03 csm_setup [45]: INFO [get_consul_config] Fetching consul endpoint : http://cortx-consul-server:8500
2022-07-18 07:43:03 csm_setup [45]: INFO [copy_base_configs] Copying Csm base configurations to destination indices
2022-07-18 07:43:05 csm_setup [45]: INFO [_prepare_and_validate_confstore_keys] Post Install: Validating required configuration.
2022-07-18 07:43:05 csm_setup [45]: INFO [_validate_conf_store_keys] Validating confstore keys: ['cortx>common>security>ssl_certificate']
2022-07-18 07:43:05 csm_setup [45]: INFO [set_ssl_certificate] Post Install: Setting SSL certificate path: /etc/cortx/solution/ssl/s3.seagate.com.pem
2022-07-18 07:43:05 csm_setup [45]: INFO [set_logpath] Post Install: Setting log path: /etc/cortx/log/csm
2022-07-18 07:43:05 csm_setup [45]: INFO [create] Post Install: Creating CSM configuration file on required location.
2022-07-18 07:43:05 csm_setup [45]: INFO [execute] Setup: Successfully passed Post Install phase.
2022-07-18 07:43:08 csm_setup [50]: INFO [execute] Setup: Initiating Prepare phase.
2022-07-18 07:43:08 csm_setup [50]: INFO [load_csm_config_indices] Loading CSM configuration
2022-07-18 07:43:08 csm_setup [50]: INFO [get_consul_config] Fetching consul endpoint : http://cortx-consul-server:8500
2022-07-18 07:43:08 csm_setup [50]: INFO [_prepare_and_validate_confstore_keys] Prepare: Validating required configuration.
2022-07-18 07:43:08 csm_setup [50]: INFO [_validate_conf_store_keys] Validating confstore keys: ['node>e0f8c65955d4e5172d56c897c7421951>hostname', 'node>e0f8c65955d4e5172d56c897c7421951>cluster_id', 'cortx>external>consul>secret']
2022-07-18 07:43:08 csm_setup [50]: INFO [_set_secret_string_for_decryption] Prepare: Setting decryption keys for CSM and S3
2022-07-18 07:43:08 csm_setup [50]: INFO [_set_cluster_id] Prepare: Setting up cluster id
2022-07-18 07:43:08 csm_setup [50]: INFO [_set_db_host_addr] Prepare: Setting database host address
2022-07-18 07:43:08 csm_setup [50]: INFO [get_consul_config] Fetching consul endpoint : http://cortx-consul-server:8500
2022-07-18 07:43:08 csm_setup [50]: INFO [create] Prepare: Creating configuration file on required location.
2022-07-18 07:43:08 csm_setup [50]: INFO [execute] Setup: Successfully passed Prepare phase.
2022-07-18 07:43:11 csm_setup [55]: INFO [execute] Setup: Initiating Config phase.
2022-07-18 07:43:11 csm_setup [55]: INFO [load_csm_config_indices] Loading CSM configuration
2022-07-18 07:43:11 csm_setup [55]: INFO [get_consul_config] Fetching consul endpoint : http://cortx-consul-server:8500
2022-07-18 07:43:11 csm_setup [55]: INFO [execute] Config: Force flag: False
2022-07-18 07:43:11 csm_setup [55]: INFO [_prepare_and_validate_confstore_keys] Config: Validating required configuration.
2022-07-18 07:43:11 csm_setup [55]: INFO [_validate_conf_store_keys] Validating confstore keys: ['node>e0f8c65955d4e5172d56c897c7421951>cluster_id', 'cortx>rgw>auth_user', 'cortx>rgw>auth_admin', 'cortx>rgw>auth_secret']
2022-07-18 07:43:11 csm_setup [55]: INFO [_set_csm_endpoint] Config: Setting CSM endpoint in configuration.
2022-07-18 07:43:11 csm_setup [55]: INFO [_set_s3_info] Config: Setting S3 information in configuration
2022-07-18 07:43:11 csm_setup [55]: INFO [_set_s3_endpoints] Config: Setting S3 endpoints in configuration
2022-07-18 07:43:11 csm_setup [55]: INFO [set_hax_endpoint] Config: Setting HAX endpoints in configuration
2022-07-18 07:43:11 csm_setup [55]: INFO [execute] Config: Creating topics for various CSM functionalities.
2022-07-18 07:43:11 csm_setup [55]: INFO [_create_topics] Config: Creating topics for communication channel.
2022-07-18 07:43:11 csm_setup [55]: INFO [_create_topics] Config: Connecting to message bus using endpoint :['tcp://cortx-kafka:9092']
2022-07-18 07:43:11 csm_setup [55]: INFO [init] MessageBus initialized as kafka
2022-07-18 07:43:11 csm_setup [55]: INFO [_create_perf_stat_topic] Config: Creating performance statistics topic
2022-07-18 07:43:11 csm_setup [55]: INFO [_create_perf_stat_topic] Config: Registering message_type:perf_stat
2022-07-18 07:43:12 csm_setup [55]: INFO [set_message_type_expire] Successfully updated perf_stat with new configuration.
2022-07-18 07:43:12 csm_setup [55]: INFO [_create_cluster_stop_topic] Config: Creating cluster stop topic
2022-07-18 07:43:12 csm_setup [55]: INFO [_create_cluster_stop_topic] Config: Registering message_type:cluster_stop
2022-07-18 07:43:14 csm_setup [55]: INFO [set_message_type_expire] Successfully updated cluster_stop with new configuration.
2022-07-18 07:43:14 csm_setup [55]: INFO [_create_cluster_admin] Creating cluster admin account
2022-07-18 07:43:14 csm_setup [55]: INFO [_create_cluster_admin] Set Cortx admin credentials in config
2022-07-18 07:43:14 csm_setup [55]: INFO [_create_cluster_admin] Creating cluster admin: cortxadmin
2022-07-18 07:43:15 csm_setup [55]: INFO [create] Config: Creating CSM configuration file on required location.
2022-07-18 07:43:15 csm_setup [55]: INFO [execute] Setup: Successfully passed Config phase.
2022-07-18 07:43:19 csm_setup [74]: INFO [execute] Setup: Initiating Init phase.
2022-07-18 07:43:19 csm_setup [74]: INFO [execute] Setup: Successfully passed Init phase.

**Review by pawankumarsrivastava — APPROVED** [review]

File: csm/core/controllers/unsupported_features.py (hunk 12 above)
Comment (abhijit1patil) [reply csm/core/controllers/unsupported_features.py]: updated

**Review by rohitkolapkar — APPROVED** [review]

**Pull request merged**